apps/token-manager/app/src/hooks.js

```diff
@@ -0,0 +1,82 @@
+import { useRef, useEffect, useCallback, useState } from 'react'
+
+export function useClickOutside(cb, existingRef) {
+  const ref = existingRef || useRef()
+  const handleClick = useCallback(
+    e => {
+      if (!ref.current.contains(e.target)) {
+        cb()
+      }
+    },
+    [cb, ref]
+  )
+
+  useEffect(() => {
+    document.addEventListener('click', handleClick, true)
+    return () => {
+      document.removeEventListener('click', handleClick, true)
+    }
+  }, [handleClick])
+
+  return { ref }
+}
+
+export function useOnBlur(cb, existingRef) {
+  const ref = existingRef || useRef()
+  const handleBlur = useCallback(
+    e => {
+      if (!ref.current.contains(e.relatedTarget)) {
+        cb()
+      }
+    },
+    [cb, ref]
+  )
+
+  return { ref, handleBlur }
+}
+
+export function useArrowKeysFocus(query, containerRef = useRef()) {
+  const [index, setIndex] = useState(-1)
```

Comment:
I'd suggest adding a comment or renaming `index` to something like `highlightedIndex`.